### PR TITLE
Added Deconstruct building command

### PR DESCRIPTION
**Pull request by Blaarkies**

### CODE REVIEW

**Comment by PetrusOrion** [comment]
In the game round processor, I suggest you update the error message for invalid command types so it will show that you can deconstruct a building when the code throws an invalid command.

File: game-rules.md [hunk 1]
```diff
@@ -75,29 +72,31 @@ The building will be a lower-case letter if it is still under construction and i
...
-* Buildings that are busy being built will not be hit by missles travelling over it.
-
+* Buildings that are busy being built will not be hit by missiles or lightning over it.
 * The energy cost of a building is deducted from a player's energy total once the command to build a building has been successful.
 * Trying to place a building without enough energy for that building will result in a **'Do nothing'** command.
-* Once a building is destroyed, another building can be placed at that location without penalty.
-* Only one building of any type can occupy a location on the map.
+* Once a building is destroyed, another building can be placed at that location.
+* Only one building can occupy a location on the map.
+* Damaging enemy buildings will award score to you, depending on the building type.
```
Comment (RohanChhipa) [inline game-rules.md]: award **a** score

File: game-rules.md [hunk 2]
```diff
@@ -75,29 +72,31 @@ The building will be a lower-case letter if it is still under construction and i
...
+* Constructing new buildings will award score to you, depending on the building type.
 
 ### Building Types:
-**The player can build one of 3 different types of buildings**
+**The player can build one of 4 different types of buildings**
 
 #### Defence Building:
-A defensive building, that is tougher than any other building type with no offensive capabilities.
-Able to take more hits than any other building, these buildings are great for soaking up damage while your offensive buildings fire from safety behind them.
+A defensive building, tougher than any other building type with no offensive capabilities.
```
Comment (RohanChhipa) [inline game-rules.md]: building **is** tougher

File: game-rules.md [hunk 3]
```diff
@@ -75,29 +72,31 @@ The building will be a lower-case letter if it is still under construction and i
...
+It is able to take more hits than any other building, these buildings are great for soaking up damage while your offensive buildings fire from the safety behind them.
```
Comment (RohanChhipa) [inline game-rules.md]: these buildings are great for soaking up damage while your **offensive buildings fire safely from behind them**

File: game-rules.md [hunk 4]
```diff
@@ -107,20 +106,21 @@ Able to take more hits than any other building, these buildings are great for so
 * Under construction character: **d**
 
 #### Attack Building:
-The main offensive building you can build that fires straight line missiles toward the opposing side. This will be the main way to damage the opposing player's buildings and base.
-The attack building has a firing rate, meaning it will fire a missile every **{Fire rate}** turns after it has been successfully built. 
+An offensive building, fires straight line missiles toward the opposing side. This is the primary method of damaging the opposing player's buildings and home base.
```
Comment (RohanChhipa) [inline game-rules.md]: **building fires**

**Pull request merged**